app/components/name-generator.js

```diff
@@ -0,0 +1,24 @@
+import Ember from 'ember';
+import _number from 'lodash/number';
+
+var beginning = ['A', 'Be', 'De', 'El', 'Fa', 'Jo', 'Ki', 'La', 'Ma', 'Na', 'O', 'Pa', 'Re', 'Si', 'Ta', 'Va'];
+var middle = ['bar', 'ched', 'dell', 'far'];
+var last = ['a','ac', 'ai', 'al', 'am', 'an', 'ar', 'ea', 'el', 'er', 'ess', 'elt', 'ic','id', 'il', 'in', 'is', 'of', 'us'];
+
+export default Ember.Component.extend({
+  actions: {
+    newName() {
+        this.setName();
+    }
+  },
+  generateName: function () {
```

Comment:
You can apply ES6 syntax here (method properties):
```
generateName() {

}
```
as you did with `newName()`.